File: examples/python/cdktf/stacks/iam/stacks.py
from cdk_organizer.terraform.stack_group import StackGroup
from stacks.config import EnvironmentStackGroupConfig
from stacks.storage.stacks import StorageStackGroup
from templates.iam_role_stack import IamRoleStack, IamRoleStackParameters


class IamStackGroup(StackGroup[EnvironmentStackGroupConfig]):
    def _load_stacks(self) -> None:
        IamRoleStack(
            self.app,
            self.get_stack_name("role"),
            stack_group=self,
            parameters=IamRoleStackParameters(
                bucket=self.resolve_group(StorageStackGroup).bucket,
                account_id=self.data.account
            )
        )

File: examples/typescript/cdktf/stacks/iam/stacks.ts
import { StackGroup } from '@awslv/cdktf-organizer';
import { RoleStack } from '../../templates/role-stack';
import { StorageStackGroup } from '../storage/stacks';

export class IamStackGroup extends StackGroup {
  override async loadStacks(): Promise<void> {
    const storageGroup = await this.resolveGroup(StorageStackGroup);

    new RoleStack(this, this.app, this.getStackName('role'), {
      bucket: storageGroup.bucket,
      accountId: this.data.account,
    });
  }
}
